database/sqlite/sqlite_context.c

```diff
@@ -130,17 +130,19 @@ void sql_close_context_database(void)
 void ctx_get_chart_list(uuid_t *host_uuid, void (*dict_cb)(SQL_CHART_DATA *, void *), void *data)
 {
     int rc;
-    sqlite3_stmt *res = NULL;
+    static __thread sqlite3_stmt *res = NULL;
 
     if (unlikely(!host_uuid)) {
        internal_error(true, "Requesting context chart list without host_id");
        return;
     }
 
-    rc = sqlite3_prepare_v2(db_context_meta, CTX_GET_CHART_LIST, -1, &res, 0);
-    if (unlikely(rc != SQLITE_OK)) {
-        error_report("Failed to prepare statement to fetch chart list");
-        return;
+    if (unlikely(!res)) {
+        rc = prepare_statement(db_context_meta, CTX_GET_CHART_LIST, &res);
+        if (rc != SQLITE_OK) {
+            error_report("Failed to prepare statement to fetch chart list");
```

Comment:
The PR is working as expected, but considering that we have the common sentence `Failed to prepare statement to fetch`, I will suggest to use it as a `define`, but considering that can be extended for code already merged, if you accept the suggestion, are you going to also extend for merged code with this PR?

Reply:
The highlighted part is a part of the sentence. What do you mean by _use it as a define_? Can you show an example code?

Reply:
I mean to call function `error_report("%s to fetch chart list", VARIABLE_NAME);` instead to repeat it for all calls, we would have something like `CTX_GET_CHART_LIST` two lines above. This will force compile to centralize the sentence in an unique place.

Reply:
We can refine later the error / info messages